### ISSUE TITLE
CI/CD Cluster (AWS) Does NOT allow batch/v1beta1 cronjobs to be deployed

### ISSUE THREAD

**Issue by doks5** [vmware/versatile-data-kit#1612]
**Describe the bug**
Currently, all data jobs that are deployed in the CI/CD cluster (AWS) use the `batch/v1` Cronjob API irrespective of what template is used. This can be verified using kubectl and issuing the `kubectl explain cronjobs` command. The result printed to the terminal is
```bash
KIND:     CronJob
VERSION:  batch/v1

DESCRIPTION:
     CronJob represents the configuration of a single cron job.

FIELDS:
   apiVersion	<string>
     APIVersion defines the versioned schema of this representation of an
     object. Servers should convert recognized schemas to the latest internal
     value, and may reject unrecognized values. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources

   kind	<string>
     Kind is a string value representing the REST resource this object
     represents. Servers may infer this from the endpoint the client submits
     requests to. Cannot be updated. In CamelCase. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds

   metadata	<Object>
     Standard object's metadata. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata

   spec	<Object>
     Specification of the desired behavior of a cron job, including the
     schedule. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

   status	<Object>
     Current status of a cron job. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
```

As per the above, the k8s server converts every cronjob object to the `batch/v1` API schema, irrespective of what API is used to create it. As a result, any Control Service integration test that relies on deploying a job with the `batch/v1beta1` API, would fail.

It is also possible to verify it by running **DataJobDeploymentCrudIT** integration test from a local environment connected to the CI/CD cluster.

**Steps To Reproduce**
1. Open the **DataJobDeploymentCrudIT** integration test in an IDE
2. Configure the IDE to use the CI/CD cluster
3. Place a break point somewhere in the **testDataJobDeploymentCrud** test case, preferably before the part, where the deployment is deleted.
4. Run the integration test in `debug` mode.
5. Check the logs for the name of the test data job created and deployed by the integration test.
6. Use kubectl to check the cronjob of the data job in the CI/CD cluster. Command is: `kubectl get cronjob <job-name> -oyaml`
7. Check the output and compare it with the logs of the integration test.

**Expected behavior**
Both `batch/v1beta1` and `batch/v1` cron jobs can be deployed in the CI/CD cluster.

**Additional context**
To reproduce the behaviour, one needs access to the CI/CD cluster.

**Comment by doks5** [comment]
I see two approaches that can be taken here. Either to somehow enable beta API cronjobs to be created in the cluster, or do a reverse-switch, i.e., the default version to be `batch/v1` and the flag to be set to enable `batch/v1beta1` to unblock the CI/CD.
Any thoughts on this, @tozka, @murphp15?

**Comment by murphp15** [comment]
Have you tested it on a local kind cluster, does the same issue happen there?

**Comment by tozka** [comment]
Yes, our CICD uses 1.22 version of kubernetes. At this point v1beta1 is officially deprecated in favour of v1 version of Cron. So it's storage version is set to v1. That means any Cron resource being created (or updated) will automatically become v1. 
It's explained here: 

https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#writing-reading-and-updating-versioned-customresourcedefinition-objects

I think we should read carefully 
https://kubernetes.io/docs/reference/using-api/deprecation-policy/  as well in order to see how to handle both Cron v1 and batch/v1beta1 properly. 


In order to test we might need an older version of Kubernetes. You can do that locally with kind. For automated testing in the CICD I am not sure I want us to create another Kubernetes cluster just for this feature especially since v1beta1 would be removed in 3 minor versions (1.26. probably) of Kubernetes.. So we might forgo end to end test here unless someone comes with some cheap idea.

**Comment by doks5** [comment]
Judging by the documentation, the v1beta1 API was dropped in 1.25, https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-25.

**Comment by doks5** [comment]
So, @tozka, @murphp15, can we then consider the comments in #1580, requesting a new integration test to be introduced, as irrelevant as long as the change is covered with unit tests and instructions how to test it locally are added to the PR?

**Comment by murphp15** [comment]
Yeah makes sense!

**Comment by sabadzhiev** [comment]
Triaged. This is fixed.
